lockfile/src/javascript.rs

```diff
@@ -263,6 +267,69 @@ impl Parse for YarnLock {
     }
 }
 
+pub struct Pnpm;
+
+impl Parse for Pnpm {
+    /// Parses `pnpm-lock.yaml` files into a vec of packages.
+    fn parse(&self, data: &str) -> anyhow::Result<Vec<Package>> {
+        let lockfile: PnpmLock = serde_yaml::from_str(data)?;
+        lockfile.packages()
+    }
+
+    fn is_path_lockfile(&self, path: &Path) -> bool {
+        path.file_name() == Some(OsStr::new("pnpm-lock.yaml"))
+    }
+
+    fn is_path_manifest(&self, path: &Path) -> bool {
+        path.file_name() == Some(OsStr::new("package.json"))
+    }
+
+    #[cfg(feature = "generator")]
+    fn generator(&self) -> Option<&'static dyn Generator> {
+        Some(&PnpmGenerator)
+    }
+}
+
+/// `pnpm-lock.yaml` structure.
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct PnpmLock {
+    #[serde(rename = "lockfileVersion")]
+    _lockfile_version: String,
+    #[serde(default)]
+    packages: HashMap<String, serde_yaml::Value>,
+}
+
+impl PnpmLock {
+    /// Get all packages in the lockfile.
+    fn packages(&self) -> anyhow::Result<Vec<Package>> {
+        let mut packages = Vec::new();
+
+        // Convert `/@org/package@1.2.3(test@3.2.1)` to package.
+        for name in self.packages.keys() {
+            // Strip `/` prefix.
+            let name = &name[1..];
```

Comment:
It's probably safer to confirm that `name[0]` is a `/` before stripping it. Something like:

```rust
let name = name.strip_prefix("/").ok_or_else(|| anyhow!("Invalid package: {name}"))?;
```